packages/web-embed-api/README.md

```diff
@@ -91,11 +91,22 @@ Set a callback to be called when the user abandons the flow before finalizing it
 
 Note that this is only possible if your style set defines a close button.
 
-#### redirect `(url: string) => void`
+#### redirect `(url: string) => { cancel?: boolean, customUrl?: string } | undefined`
```

Comment:
Not sure I see the value in having both a `cancel` and a `customUrl` prop.

One reason is: you can provide an inconsistent state like `{ cancel: true, customUrl: "https://example.com" }` ... in this case, would you cancel? Or would you redirect?

IMO, we should just support handling the boolean for cancel only. If cancelled - it's the developer's responsibility to redirect (it's a 1 line for them to do).

I do like the object return - it will allow us to more easily extend further if we need to later.

What do you think of that simplification?

Reply:
In case of having both cancel and redirect, I would cancel and ignore the redirect. But as for the bigger point, I'm also not sure if the is much value in allowing the user-provided callback to return a custom url.  The only advantage I can see is - we have a small amount of custom logic in how we redirect in [onRedirectMessage](https://github.com/formsort/oss/blob/3fff9e664cb6f78f35b3a90a93bf57b2f3c8c6e1/packages/web-embed-api/src/index.ts#L67) that the users don't have access to. But I think that if they are at the point of wanting to use a custom url, they would probably have their own way of redirecting as well. TLDR I'll make the simplification.